src/vm/node_modules/VM.js

```diff
@@ -11633,15 +11634,27 @@ function getAllDatasets(vmobj)
 {
     var datasets = [];
     var disk;
+    var topmost_dataset;
 
     if (vmobj.hasOwnProperty('zfs_filesystem')) {
-        datasets.push(vmobj.zfs_filesystem);
+        topmost_dataset = vmobj.zfs_filesystem;
+        datasets.push(topmost_dataset);
     }
 
+    // Put a trailing slash to make child-dataset detection easier.
+    topmost_dataset += '/';
+
     for (disk in vmobj.disks) {
+        var disk_dataset;
+
         disk = vmobj.disks[disk];
         if (disk.hasOwnProperty('zfs_filesystem')) {
-            datasets.push(disk.zfs_filesystem);
+            disk_dataset = disk.zfs_filesystem;
+            // Don't push if child of top-level zfs_filesystem.
```

Comment:
Turn this comment around. Something like this:

```
// Child datasets of the zone root are automatically included because of -r (recursive).
// If the dataset name is not a child of the zone root, then add it to the list
// of datasets we need to transfer.
```

Reply:
Will fix.  Please mark this resolved when it is to your satisfaction.